File: android/src/main/java/dragon3/common/constant/ArmorType.java
/*
 * Created on 2004/06/27
 */
package dragon3.common.constant;

import java.util.LinkedHashMap;
import java.util.Map;

import lombok.Getter;

/**
 * @author k-saito
 */
public enum ArmorType {
	
	NONE("無"),
	LITE("軽量"),
	HEAVY("重量"),
	;
	
	@Getter private String text;
	
	ArmorType(String text) {
		this.text = text;
	}
	
	public static Map<String, String> createMap() {
		Map<String, String> idAndText = new LinkedHashMap<>();
		for (ArmorType a : values()) {	
			idAndText.put(a.name(), a.getText());
		}
		return idAndText;
	}

}

File: core/src/main/java/dragon3/common/constant/ArmorType.kt
/*
 * Created on 2004/06/27
 */
package dragon3.common.constant

import java.util.LinkedHashMap

/**
 * @author k-saito
 */
enum class ArmorType private constructor(
        val text: String) {

    NONE("無"),
    LITE("軽量"),
    HEAVY("重量");


    companion object {

        fun createMap(): Map<String, String> {
            val idAndText = LinkedHashMap<String, String>()
            for (a in values()) {
                idAndText.put(a.name, a.text)
            }
            return idAndText
        }
    }

}
